kafka/offset_tracker.go

```diff
@@ -44,6 +55,14 @@ func (k *KafkaOffsetTracker) run() {
 			//log.Printf("waiting for url %s to process, queue_len %d", kmsg.GetURLPath(), len(k.ch))
 			time.Sleep(100 * time.Microsecond)
 		}
-		k.source.CommitOffsets(kmsg)
+
+		if isUpdated, err := k.source.CommitOffsets(kmsg); isUpdated && err == nil && k.source.offMonitor.SourceSinkMonitorEnabled {
+			msg := kmsg.GetRawMsg()
+			metrics.Ingest(metrics.Metric{
+				Type:  metrics.Offset,
+				Name:  "sink_offset" + "." + k.source.conf.ConsumerGroupName + "." + msg.Topic + "." + strconv.Itoa(int(msg.Partition)),
+				Value: msg.Offset,
+			})
```

Comment:
we should ideally move the logic of reading flag and pushing metric to offset monitor itself and simply call from here